opensrp-app/src/main/java/org/smartregister/repository/TaskRepository.java

```diff
@@ -684,4 +684,48 @@ public int getUnsyncedCreatedTasksAndTaskStatusCount() {
         String query = "SELECT * FROM " + TASK_TABLE + " WHERE " + GROUP_ID + " = ?";
         return getTasks(query, new String[]{jurisdictionId});
     }
+
+    public List<String> getEntityIdsWithDuplicateTasks() {
+        List<String> entityIds = new ArrayList<>();
+        android.database.Cursor cursor = null;
+        String query = "SELECT\n" +
+                "    DISTINCT(for)\n" +
+                "FROM\n" +
+                "    task\n" +
+                "GROUP BY\n" +
+                "    plan_id, for, code\n" +
+                "HAVING \n" +
+                "    COUNT(*) > 1";
+
+        cursor = getReadableDatabase().rawQuery(query, new String[]{});
+
+        while (cursor.moveToNext()) {
+            entityIds.add(cursor.getString(0));
+        }
+        return entityIds;
+    }
+
+    public Set<Task> getDuplicateTasksForEntity(String entityId) {
+        String query = "SELECT t1.*\n" +
+                "FROM task t1\n" +
+                "JOIN (SELECT\n" +
+                "    _id, plan_id, for, code, COUNT(*) as count\n" +
+                "FROM\n" +
+                "    task\n" +
+                "GROUP BY\n" +
+                "    plan_id, for, code\n" +
+                "HAVING \n" +
+                "    COUNT(*) > 1) t2\n" +
+                "ON t1.plan_id = t2.plan_id\n" +
+                "AND t1.for = t2.for\n" +
+                "AND t1.code = t2.code\n" +
+                "AND t1.for = ? \n" +
+                "ORDER BY t1.for";
```

Comment:
can we remove \n for readability of the query, you can use space at the end/start of each line